docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/How_Tos/Extend_CMS_Interface.md

```diff
@@ -204,26 +204,443 @@ To make the actions more user-friendly you can also use alternating buttons as
+services are. Rather, new components are composed using [higher order components](https://facebook.github.io/react/docs/higher-order-components.html).
+This has the inherent advantage of allowing all thidparty code to have an influence
+over the behaviour, state, and UI of a component.
+
+#### A simple higher order component
 
+Using our example above, let's create a customised `PhotoItem` that allows a badge,
+perhaps indicating that it is new to the gallery.
+
+```js
+const enhancedPhoto = (PhotoItem) => (props) => {
+	const badge = props.isNew ? 
+	  <div className="badge">New!</div> : 
+	  null;
+
+	return (
+		<div>
+			{badge}
+			<PhotoItem {...props} />
+		</div>
+	);
 }
 
-export default MyComponent;
+const EnhancedPhotoItem = enhancedPhoto(PhotoItem);
+
+<EnhancedPhotoItem isNew={true} size={300} />
+```
+
+Alternatively, this component could be expressed with an ES6 class, rather than a simple
+function.
+
+```js
+const enhancedPhoto = (PhotoItem) => {
+	return class EnhancedPhotoItem extends React.Component {
+		render() {
+			const badge = this.props.isNew ? 
+			  <div className="badge">New!</div> : 
+			  null;
+
+			return (
+				<div>
+					{badge}
+					<PhotoItem {...this.props} />
+				</div>
+			);
+
+		}
+	}
+}
 ```
 
-That's how you create a SilverStripe React component!
+When components are stateless, using a simple function in lieu of a class is recommended.
+
+#### Using the injector to customise a core component
+
+Let's make a more awesome text field. Because the `TextField` component is fetched 
+through the injector, we can override it and augment it with our own functionality.
+
+In this example, we'll add a simple character count below the text field.
+
+First, let's create our higher order component.
+__my-module/js/components/CharacterCounter.js__
+```js
+import React from 'react';
+
+const CharacterCounter = (TextField) => (props) => {
+    return (
+        <div>
+            <TextField {...props} />
+            <small>Character count: {props.value.length}</small>
+        </div>
+    );
+}
+
+export default CharacterCounter;
+```
+
+Now let's add this higher order component to the injector. 
+
+__my-module/js/main.js__
+```js
+import Injector from 'lib/Injector';
+import CharacterCounter from './components/CharacterCounter';
+
+Injector.transform('my-transformation', (update) => {
+  update('TextField', CharacterCounter);
+});
+```
+
+Much like the configuration layer, we need to specify a name for this transformation. This
+will help other modules negotiate their priority over the injector in relation to yours.
+
+The second parameter of the `transform` argument is a callback which receives a `update()` function
+that allows you to mutate the DI container with a wrapper for the component. Remember, this function does not _replace_
+the component -- it enhances it with new functionality.
+
+The last thing we'll have to do is transpile our code and load the resulting bundle file
```

Comment:
I think we can do this starting point later on after the webpack2 upgrade work has been done. As that contains a lot of the boilerplate work needed to hit the ground running for webpack in modules.